src/main/scala/it/pagopa/interop/catalogprocess/api/impl/ProcessApiServiceImpl.scala

```diff
@@ -318,6 +318,23 @@ final case class ProcessApiServiceImpl(
     }
   }
 
+  private def applyVisibilityToEService(
+    catalogItem: CatalogItem,
+    organizationId: UUID,
+    role: String
+  ): Future[CatalogItem] = {
+    if (Seq(ADMIN_ROLE, API_ROLE).contains(role) && catalogItem.producerId == organizationId)
+      Future.successful(catalogItem)
+    else {
+      catalogItem match {
+        case CatalogItem(_, _, _, _, _, _, descriptors, _, _, _) if descriptors.forall(_.state == Draft) =>
+          Future.failed(EServiceNotFound(catalogItem.id.toString))
+        case CatalogItem(_, _, _, _, _, _, descriptors, _, _, _)                                         =>
+          Future.successful(catalogItem.copy(descriptors = descriptors.filterNot(_.state == Draft)))
+      }
+    }
```

Comment:
we could probably avoid the pattern matching
```suggestion
    else if (catalogItem.descriptors.forall(_.state == Draft))
        Future.failed(EServiceNotFound(catalogItem.id.toString))
    else
        Future.successful(catalogItem.copy(descriptors = descriptors.filterNot(_.state == Draft)))
    }
```